backend/app/routers/alerts.py

```
@@ -110,10 +112,27 @@ async def create_alert(req: CreateAlertRequest) -> AlertResponse:
         try:
             now = _now_utc_iso()
+            rule_dict = req.rule.model_dump()
+            
+            # Extract behavior from NLP if nl is provided and behavior is not already set
+            if req.nl and not rule_dict.get("behavior"):
+                try:
+                    parsed = parse_nl_with_llm(req.nl)
+                    # Extract action from parsed NLP result and use it as behavior
+                    if parsed.get("action"):
+                        rule_dict["behavior"] = parsed.get("action")
+                except Exception:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)